core/txpool/legacypool/list.go

```diff
@@ -160,13 +160,12 @@ func (m *sortedMap) Cap(threshold int) types.Transactions {
 	// Otherwise gather and drop the highest nonce'd transactions
 	var drops types.Transactions
 
-	sort.Sort(*m.index)
+	sort.Sort(*m.index) // the sorted array is a heap, so no need to reheap after deleting tail items
```

Comment:
I suspect an even bigger gain would be to replace this with 
```suggestion
	slices.Sort(*m.index) // the sorted array is a heap, so no need to reheap after deleting tail items
```